src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -365,7 +365,14 @@ protected override async Task OnDeleteAsync(Uri resourceUri, CancellationToken c
         /// <returns>The blob uri.</returns>
         public override Uri GetUri(string name)
         {
-            return new Uri(_directory.Uri, name);
+            var baseUri = _directory.Uri.AbsoluteUri;
+
+            if (baseUri.EndsWith("/"))
+            {
+                return new Uri($"{baseUri}{name}", UriKind.Absolute);
+            }
+
+            return new Uri($"{baseUri}/{name}", UriKind.Absolute);
```

Comment:
This is a minor bug fix.  The `Uri` constructor overload that takes a `Uri` object as its first argument wouldn't use the absolute URI.  The result was when saving registration cursor, this would be logged:

    SaveAsync https://<account>.blob.core.windows.net/cursor.json

With this change, it now logs the container:

    SaveAsync https://<account>.blob.core.windows.net/v3-registration3/cursor.json